packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1895,17 +2469,54 @@ class _RouteEntry {
   }
 
   static final _RouteEntryPredicate isPresentPredicate = (_RouteEntry entry) => entry.isPresent;
+  static final _RouteEntryPredicate suitableForTransitionAnimationPredicate = (_RouteEntry entry) => entry.suitableForTransitionAnimation;
   static final _RouteEntryPredicate willBePresentPredicate = (_RouteEntry entry) => entry.willBePresent;
 
   static _RouteEntryPredicate isRoutePredicate(Route<dynamic> route) {
     return (_RouteEntry entry) => entry.route == route;
   }
+
+  @override
+  bool get isEntering => currentState == _RouteLifecycle.staging;
+
+  @override
+  void markForPush() {
+    assert(isEntering && !_debugWaitingForExitDecision);
```

Comment:
The asserts in the mark* methods should have an error text telling people what's wrong.